x-pack/plugins/ingest_manager/common/types/models/datasource.ts

```diff
@@ -5,31 +5,23 @@
  */
 
 export interface NewDatasource {
-  name: string;
-  namespace?: string;
-  config_id: string;
-  enabled: boolean;
+  enabled?: boolean;
+  title?: string;
   package?: {
-    assets: Array<{
-      id: string;
-      type: string;
-    }>;
-    description: string;
     name: string;
-    title: string;
     version: string;
   };
-  output_id: string;
+  namespace?: string;
+  use_output: string;
   inputs: Array<{
     type: string;
-    enabled: boolean;
     processors?: string[];
     streams: Array<{
-      id: string;
-      enabled: boolean;
-      dataset: string;
-      processors?: string[];
-      config?: Record<string, any>;
+      id?: string;
+      enabled?: boolean;
+      dataset?: string;
+      metricset?: string;
+      paths?: string[];
```

Comment:
There are examples of fields that don't need to exist in the saved object schema. `metricset` and `paths` are _not applicable to all streams_, that's why we have a `config` property for arbitrary stream config fields that will then be flattened into the top-level streams object at the business logic layer of converting SO -> agent config.

Reply:
Yeah, I was wondering if this wasn't perhaps an `A | B | C` type and if we needed to be that involved in the Agent configs